carto/analysis.py

```diff
@@ -0,0 +1,729 @@
+        self.table = self.definition.get('table')
+        self.scheduling_time = self.definition.get('scheduling_time')
+        if not self.time and self.scheduling_time:
+            self.time = _timefromiso(self.scheduling_time)
+
+    def _fetch_executions(self):
+        if self._executions_pending:
+            self._executions_pending = False
+            execs = self._context.api.schedule_executions(
+                self.job_id, self.schedule_id
+            )
+            # TODO: is this necessary?
+            # (does Nomad API guarantee the result order? )
+            execs.sort(key=lambda e: e['time'])
+            self._executions = [
+                Execution(
+                    self._context, self.job_id, self.schedule_id, exec
+                )
+                for exec in execs
+            ]
+
+    def update(self):
+        if self._need_to_update():
+                self.token = self._context.tokens.schedule_token(
+                    self.job_id, self.schedule_id
+                )
+                new_raw_status = self._context.api.schedule_status(
+                    self.job_id, self.schedule_id
+                )
+                if new_raw_status != self._raw_status:  # FIXME: ok??
+                    self._set_raw_status(new_raw_status)
+                    self._executions_pending = True
+                    self._updated_now()
+                    return True
+                self._updated_now()
+        return False
+
+    def is_periodic(self):
+        return self.schedule is not None
+
+    def is_triggered(self):
+        return self.type == 'trigger'
+
+    def info(self):
+        # TODO: identify schedule by user, time of creation
+        # option to include job name
+        # note: user will only be available when adding db metadata to api
+        # time could be obtained from "SubmitTime":1548327600000512493
+        return self._info_prefix() + self._info_suffix()
+
+    def _info_prefix(self):
+        return '{id} {t}'.format(
+            id=colored('bold', self.token),
+            t=_timetoiso(self.time)
+        )
+
+    def _info_suffix(self):
+        if self.is_periodic():
+            return ' ({s})'.format(s=colored('pink', self.schedule))
+        elif self.is_triggered():
+            return ' (on changed {t})'.format(t=colored('pink', self.table))
+        else:
+            return ''
+
+    def show(self, details=False):
+        color = _status_color(self.status)
+        self._context.output.print(
+            self._info_prefix()
+            + ' : ' + colored(color, self.status)
+            + ' : ' + self.summary_info()
+            + self._info_suffix()
+        )
+        if details:
+            self._context.output.print(self.details())
+
+    def details(self):
+        return ''
+
+    def summary_info(self):
+        if self.summary is None:
+            return ''
+        if self.type == 'now':
+            return 'C:{c} S:{s} F:{f} L:{l} Q:{q} R:{r}'.format(
```

Comment:
Sorry for being an uninvited guest, but how about using a cartocolor palette instead?? :)